src/main/java/org/datadog/jmxfetch/Instance.java

```diff
@@ -257,8 +271,50 @@ public Instance(
         } else {
             log.info("collect_default_jvm_metrics is false - not collecting default JVM metrics");
         }
+
+        JMXbean = createJMXBean();
+    }
+
+    private ObjectName getObjName(String domain,String instance) throws MalformedObjectNameException {
+        //return new ObjectName(domain + ":name=" + name);
+        return new ObjectName(domain + ":target_instance=" + escape(instance));
+        //return new ObjectName(domain + ":name=" + ObjectName.quote(escape(instance)));
+        //return new ObjectName(domain, "name", escape(instance));
+    }
+
+    public String escape(String inputString){
+        final String[] metaCharacters = {"*","?"};
+    
+        for (int i = 0 ; i < metaCharacters.length ; i++){
+            if(inputString.contains(metaCharacters[i])){
+                inputString = inputString.replace(metaCharacters[i],"\\"+ metaCharacters[i]);
+            }
+        }
+        inputString = "\"" + inputString + "\"";
```

Comment:
lets try to use the `ObjectName.quote` where its possible, rather than manually quoting/escaping/unquoting